dask/dataframe/tests/test_groupby.py

```diff
@@ -2094,6 +2094,41 @@ def test_series_groupby_idxmax_skipna(skipna):
     assert_eq(result_pd, result_dd)
 
 
+def test_groupby_unique():
+    rng = np.random.RandomState(42)
+    df = pd.DataFrame(
+        {"foo": rng.randint(3, size=100), "bar": rng.randint(10, size=100)}
+    )
+    ddf = dd.from_pandas(df, npartitions=10)
+
+    pd_gb = df.groupby("foo")["bar"].unique()
+    dd_gb = ddf.groupby("foo")["bar"].unique().compute()
```

Comment:
I wonder if we might not call `compute` here, but instead use `assert_eq`.  This is nice both because it avoids a bunch of custom testing code, like what you had to do below, and it also runs a bunch of other checks on the dask collection that we find valuable (sane key names, divisions, and so on).  

I suspect that you didn't do this because the order of rows may differ.  I wonder if this might be resolved by calling `set_index` or some other sort function before passing the values into `assert_eq`